src/reporting/Market.sol

```diff
@@ -0,0 +1,604 @@
+        if (block.timestamp <= reportingWindow.getEndTime()) {
+            return false;
+        }
+        return true;
+    }
+
+    function canFinalizeFork() private returns (bool) {
+        bytes32 _winningPayoutDistributionHash = MarketExtensions(controller.lookup("MarketExtensions")).getWinningPayoutDistributionHashFromFork(this);
+        if (_winningPayoutDistributionHash == bytes32(0)) {
+            return false;
+        }
+        tentativeWinningPayoutDistributionHash = _winningPayoutDistributionHash;
+        return true;
+    }
+
+    function migrateThroughAllForks() public returns (bool) {
+        // this will loop until we run out of gas, follow forks until there are no more, or have reached an active fork (which will throw)
+        while (migrateThroughOneFork()) {
+            continue;
+        }
+        return true;
+    }
+
+    // returns 0 if no move occurs, 1 if move occurred, throws if a fork not yet resolved
+    function migrateThroughOneFork() public returns (bool) {
+        if (isFinalized()) {
+            return true;
+        }
+        if (!needsMigration()) {
+            return false;
+        }
+        // only proceed if the forking market is finalized
+        require(reportingWindow.isForkingMarketFinalized());
+        Branch _currentBranch = getBranch();
+        // follow the forking market to its branch and then attach to the next reporting window on that branch
+        bytes32 _winningForkPayoutDistributionHash = _currentBranch.getForkingMarket().getFinalPayoutDistributionHash();
+        Branch _destinationBranch = _currentBranch.getChildBranch(_winningForkPayoutDistributionHash);
+        ReportingWindow _newReportingWindow = _destinationBranch.getNextReportingWindow();
+        _newReportingWindow.migrateMarketInFromNibling();
+        reportingWindow.removeMarket();
+        reportingWindow = _newReportingWindow;
+        // reset to unreported state
+        limitedReportersDisputeBondToken = DisputeBondToken(0);
+        allReportersDisputeBondToken = DisputeBondToken(0);
+        tentativeWinningPayoutDistributionHash = 0;
+        return true;
+    }
+
+    ////////
+    //////// Helpers
+    ////////
+
+    function getReportingToken(uint256[] _payoutNumerators) public returns (ReportingToken) {
+        bytes32 _payoutDistributionHash = derivePayoutDistributionHash(_payoutNumerators);
+        ReportingToken _reportingToken = reportingTokens[_payoutDistributionHash];
+        if (address(_reportingToken) == NULL_ADDRESS) {
+            _reportingToken = ReportingTokenFactory(controller.lookup("ReportingTokenFactory")).createReportingToken(controller, this, _payoutNumerators);
+            reportingTokens[_payoutDistributionHash] = _reportingToken;
+        }
+        return _reportingToken;
+    }
+
+    function fundDisputeBondWithReputation(address _bondHolder, DisputeBondToken _disputeBondToken, uint256 _bondAmount) private returns (bool) {
```

Comment:
I ended up just removing the function which did cut down size a bit since it has to come after the dispute bond creation.